internal/pkg/scalable/scaledobjects.go

```diff
@@ -0,0 +1,98 @@
+package scalable
+
+import (
+	"context"
+	"fmt"
+	kedav1alpha1 "github.com/kedacore/keda/v2/apis/keda/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes"
+	"log/slog"
+	"strconv"
+)
+
+const (
+	kedaPausedReplicasAnnotation = "autoscaling.keda.sh/paused-replicas"
+)
+
+// getScaledObjects is the getResourceFunc for Keda ScaledObjects
+func getScaledObjects(namespace string, clientset *kubernetes.Clientset, dynamicClient dynamic.Interface, ctx context.Context) ([]Workload, error) {
+	var results []Workload
+	scaledobjects, err := dynamicClient.Resource(kedav1alpha1.SchemeGroupVersion.WithResource("scaledobjects")).Namespace(namespace).List(ctx, metav1.ListOptions{TimeoutSeconds: &timeout})
+	slog.Error("error before fetching")
+	if err != nil {
+		slog.Error("error after fetching")
+		return nil, fmt.Errorf("failed to get scaledobjects: %w", err)
+	}
+	for _, item := range scaledobjects.Items {
+		slog.Error("error inside foreach")
+		so := &kedav1alpha1.ScaledObject{}
+		if err := runtime.DefaultUnstructuredConverter.FromUnstructured(item.Object, so); err != nil {
+			return nil, fmt.Errorf("failed to convert unstructured to scaledobject: %w", err)
+		}
+		results = append(results, ScaledObject{so})
+	}
+	slog.Error("no error")
+	return results, nil
+}
+
+// ScaledObject is a wrapper for keda.sh/v1alpha1.HorizontalPodAutoscaler to implement the scalableResource interface
+type ScaledObject struct {
+	*kedav1alpha1.ScaledObject
+}
+
+// SetPauseScaledObjectAnnotation will modify the value of the keda pause annotation to the one passed as a parameter
+func (s ScaledObject) SetPauseScaledObjectAnnotation(stringReplicas string) {
+	if s.Annotations == nil {
+		s.Annotations = map[string]string{}
+		s.Annotations[kedaPausedReplicasAnnotation] = stringReplicas
+	} else {
+		s.Annotations[kedaPausedReplicasAnnotation] = stringReplicas
+	}
+}
+
+// RemovePauseScaledObjectAnnotation will remove the keda pause annotation
+func (s ScaledObject) RemovePauseScaledObjectAnnotation() error {
```

Comment:
This func is pretty useless, and has a lot of unneeded overhead.
The functionality could be achived just with: (we don't need an error if it doesn't exist, we just care that is no longer exists after calling this)
``` go
// RemovePauseScaledObjectAnnotation will remove the keda pause annotation
func (s ScaledObject) RemovePauseScaledObjectAnnotation() {
    delete(s.Annotations, kedaPausedReplicasAnnotation)
} 
```

Possibly everything to do with this should be refactored to not have to neeed this funciton at all

Reply:
I didn't understand this point, do you prefer to call directly the map delete from the upscaling function without referring to this method inside the ScaledObject class?

Reply:
well, the function would only be the delete, so i would refactor the upscaling func to allow for this behavior without needing this one-liner func.

Reply:
functions deleted, used the default already implemented inside Workload Interface